mne/time_frequency/spectrum.py

```diff
@@ -1643,6 +1643,74 @@ def __init__(
         )
 
 
+def combine_spectrum(all_spectrum, weights="nave"):
+    """Merge spectral data by weighted addition.
+
+    Create a new :class:`mne.time_frequency.Spectrum` instance, using a combination of
+    the supplied instances as its data. By default, the mean (weighted by trials) is
+    used. Subtraction can be performed by passing negative weights (e.g., ``[1, -1]``).
+    Data must have the same channels and the same time instants.
+
+    Parameters
+    ----------
+    all_spectrum : list of Spectrum
+        The tfr datasets.
+    weights : list of float | str
+        The weights to apply to the data of each :class:`~mne.time_frequency.Spectrum`
+        instance, or a string describing the weighting strategy to apply: 'nave'
+        computes sum-to-one weights proportional to each object’s nave attribute;
+        'equal' weights each :class:`~mne.time_frequency.Spectrum` by
+        ``1 / len(all_spectrum)``.
+
+    Returns
+    -------
+    spectrum : Spectrum
+        The new spectral data.
+
+    Notes
+    -----
+    .. versionadded:: 1.10.0
+    """
+    spectrum = all_spectrum[0].copy()
+    if isinstance(weights, str):
+        if weights not in ("nave", "equal"):
+            raise ValueError('Weights must be a list of float, or "nave" or "equal"')
+        if weights == "nave":
+            for s_ in all_spectrum:
+                if s_.nave is None:
+                    raise ValueError(f"The 'nave' attribute is not specified for {s_}")
+            weights = np.array([e.nave for e in all_spectrum], float)
+            weights /= weights.sum()
+        else:  # == 'equal'
+            weights = [1.0 / len(all_spectrum)] * len(all_spectrum)
+    weights = np.array(weights, float)
+    if weights.ndim != 1 or weights.size != len(all_spectrum):
+        raise ValueError("Weights must be the same size as all_spectrum")
+
+    ch_names = spectrum.ch_names
+    for s_ in all_spectrum[1:]:
+        assert s_.ch_names == ch_names, ValueError(
+            f"{spectrum} and {s_} do not contain the same channels"
+        )
+        assert np.max(np.abs(s_.freqs - spectrum.freqs)) < 1e-7, ValueError(
+            f"{spectrum} and {s_} do not contain the same frequencies"
+        )
```

Comment:
This pattern doesn't work the way it seems like it should:

```pycon
In [1]: assert 1 == 2, ValueError("foo")
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[1], line 1
----> 1 assert 1 == 2, ValueError("foo")

AssertionError: foo
```
Note that the result is still an `AssertionError`, not a `ValueError`. Here I suggest removing the ValueError, but just as good would be something like `if x != y: raise ValueError(...)`.  Feel free to change to that pattern if you prefer.

```suggestion
        assert s_.ch_names == ch_names, (
            f"{spectrum} and {s_} do not contain the same channels"
        )
        assert np.max(np.abs(s_.freqs - spectrum.freqs)) < 1e-7, (
            f"{spectrum} and {s_} do not contain the same frequencies"
        )
```

Reply:
Ah okay. I copied this structure from `combine_tfr()`:
https://github.com/mne-tools/mne-python/blob/b02db17b92b3274c0ede0969503f05a7eb721c16/mne/time_frequency/tfr.py#L3805-L3811

Should I also change that?

Reply:
yeah, that's wrong / doesn't do what it looks like it's trying to do.

Reply:
Sorted both instances now.